contracts/multisig/src/state.rs

```diff
@@ -19,3 +20,5 @@ pub fn get_key(store: &dyn Storage, key_id: &KeyID) -> Result<Key, ContractError
             key_id: key_id.to_string(),
         })
 }
+
+pub const PUB_KEYS: Map<(Addr, KeyType), PublicKey> = Map::new("registered_pub_keys");
```

Comment:
in this case I would use the raw binary data for the public key, to prevent errors where KeyType and PublicKey don't match up. Then, we can define a conversion from ((addr, keytype), key) to (addr, typed_key)